### PR TITLE
Add config option to disable syncing of TODOs

### PR DESCRIPTION
**Pull request by ramblingenzyme**
To address https://github.com/lfos/calcurse/issues/55 particularly for Google Calendar syncing.

### CODE REVIEW

**Comment by watersalesman** [comment]
This does not address the issue in #55 as is. It actually breaks TODO syncing altogether by attempting to export the objects without formatting them.

Take a look at some of the filtering flags in the man page to get the right functionality. Also the new option should probably be added to the example config file and maybe the documentation.

**Comment by ramblingenzyme** [comment]
Yup, I realised that...I've now used the filter flags and added documentation.

**Comment by lfos** [comment]
How about, instead of adding a `SyncTodo` option, allowing for setting the `filter-type` value from the configuration file?

**Comment by lfos** [comment]
Maybe even allow for specifying the whole import command? Not saying that it is a good idea, just something I wanted to throw in for discussion...

**Comment by lfos** [comment]
I plan on releasing 4.3.0 in the next couple of days, by the way. If we want this feature in there, we should try to find a concensus soon...

**Comment by watersalesman** [comment]
I am in favor of the filter-type config option. Makes the script more flexible going forward (e.g. if someone wants to use basic auth but does not want to sync TODOs and recurring events, or this case where TODOs cause an error with Google Calendar).

**Comment by ramblingenzyme** [comment]
I'm happy to get on this by tomorrow when I'm a bit more free.

File: contrib/caldav/calcurse-caldav.py [hunk 1]
```diff
@@ -43,6 +43,12 @@ def die_atnode(msg, node):
         msg += etree.tostring(node).decode('utf-8')
     die(msg)
 
+def validate_sync_filter():
+    valid_sync_filter_values = {'event', 'apt', 'recur-event', 'recur-apt', 'todo', 'recur', 'cal'}
+
+    filter_items = set(sync_filter.split(','))
+
+    return filter_items - valid_sync_filter_values
```
Comment (lfos) [inline contrib/caldav/calcurse-caldav.py]: Small nit: I think this is easier to read if you make it a two-liner, initializing `valid_sync_filter_values` in the first line and computing the return value (i.e. `return set(sync_filter.split(',')) - valid_sync_filter_values`) in the second line, without an empty line inbetween.

Comment (lfos) [reply contrib/caldav/calcurse-caldav.py]: We also separate functions by two empty lines. So there is one empty line missing before the function and another one after the function.

File: contrib/caldav/calcurse-caldav.py [hunk 2]
```diff
@@ -53,13 +59,20 @@ def calcurse_wipe():
...
-                          '--format-apt=%(hash)\\n',
-                          '--format-recur-apt=%(hash)\\n',
-                          '--format-event=%(hash)\\n',
-                          '--format-recur-event=%(hash)\\n',
-                          '--format-todo=%(hash)\\n'],
-                         stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+    command = [
+        calcurse,
+        '-i', '-',
+        '--filter-type', sync_filter,
```
Comment (lfos) [inline contrib/caldav/calcurse-caldav.py]: Minor nit: This should go after, `--dump-imported` and `-q`, just before the formatting options.

File: contrib/caldav/config.sample [hunk 3]
```diff
@@ -18,6 +18,13 @@ Path = /path/to/calendar/on/the/server/
 # Enable this if you want to skip SSL certificate checks.
 InsecureSSL = No
 
+# This option allows you to filter the type of tasks synced and the valid options are
+# event,apt, recur-event, recur-apt, todo, recur, cal exactly like the command line option
```
Comment (lfos) [inline contrib/caldav/config.sample]: Nit: Missing space after `event,`. Maybe also mention that there should not be any whitespace in the comma-separated list itself.

**Comment by lfos** [comment]
I decided to not include this in 4.3.0, so there is no need to rush. I plan on releasing 4.4.0 soon after 4.3.0, though, because there is a lot of WIP on `calcurse-caldav`. The plan is to make proper support for both Google Calendar and iCloud the milestone to reach for 4.4.0.

**Comment by lfos** [comment]
Merged with some small changes to the commit message and to the comment in the sample configuration file. Thanks!

**Pull request closed without merging**